<img src="https://i.ibb.co/fxNxyFR/pexels-deepak-gautam-240163.jpg" width = 600>

# 1. Introdução

## O que são operações aritméticas?
Operações aritméticas são operações matemáticas que envolvem números e símbolos matemáticos, como adição, subtração, múltiplicação e algumas outras. Na programação, as operações aritméticas são muito importantes, pois permitem que os programas realizem cálculos matemáticos de forma rápida e eficiente.

<hr>

# 2. Operadores aritméticos

## Adição (+)
Vamos começar com o operador mais simples, o de adição! Ele é representado pelo sinal de mais (+) e é utilizado para somar dois valores. Observe o exemplo abaixo:

In [14]:
# Atribuindo o valor 3 à variável x
x = 3

# Atribuindo o valor 7 à variável y
y = 7

# Exibindo o valor da soma de x e y
print(f"A soma de x e y é: {x + y}")

A soma de x e y é: 10


## Subtração (-)
Agora, vamos falar sobre o operador de subtração. Ele é representado pelo sinal de menos (-) e é utilizado para subtrair um valor de outro. Observe o exemplo abaixo:

In [17]:
# Atribuindo o valor 2.7 à variável x
x = 2.7

# Atribuindo o valor 1.25 à variável y
y = 1.25

# Exibindo o valor da subtração de x por y
print(f"Subtração de x por y: {x - y}")

Subtração de x por y: 1.4500000000000002


## Multiplicação (\*)
Falando agora sobre o operador de multiplicação, ele é representado pelo sinal de asterisco (\*) e é utilizado para multiplicar dois valores. Observe o exemplo abaixo:

In [18]:
# Atribuindo o valor 0.5 à variável x
x = 0.5

# Atribuindo o valor 28 à variável y
y = 28

# Exibindo o valor da multiplicação de x e y
print(f"A multiplicação de x por y é: {x*y}")

A multiplicação de x por y é: 14.0


## Divisão (/)
O operador de divisão é representado pelo sinal de barra (/) e é utilizado para dividir um valor pelo outro. Observe o exemplo abaixo:

In [19]:
# Atribuindo o valor 70 à variável x
x = 70

# Atribuindo o valor 7 à variável y
y = 7

# Exibindo o valor da divisão de x por y
print(f"A divisão de x por y é: {x/y}")

A divisão de x por y é: 10.0


## Potência (\*\*)
O operador de potência, ao contrário do que muitos pensam, não é representado pelo acento circunflexo ("chapeuzinho": ^), mas sim por dois sinais de asterisco (\*\*) e é utilizado para elevar um valor a uma determinada potência. Observe o exemplo abaixo:

In [23]:
# Atribuindo o valor 5 à variável x
x = 5

# Atribuindo o valor 2 à variável y
y = 2

# Exibindo o valor de x elevado a y
print(f"O valor de x elevado a y é: {x**y}")

O valor de x elevado a y é: 25


## Divisão inteira (//)
Quando falamos de divisão inteira, falamos sobre pegar a parte inteira da divisão de um número por outro. Imagine que você queira dividir 11 por 2. O resultado é 5.5, com parte inteira 5 e parte não inteira 0.5. Dessa forma, quando utilizamos o operador de divisão inteira (//), pegamos apenas 5. Observe o exemplo abaixo: 

In [27]:
# Atribuindo o valor 11 à variável x
x = 11

# Atribuindo o valor 2 à variável y
y = 2

# Exibindo o valor da divisão inteira de x por y
print(f"Divisão inteira de x por y: {x//y}")

Divisão inteira de x por y: 5


## Resto da divisão (%)
Indo para outro operador aritmético que alguns podem não conhecer, vamos falar sobre o operador que nos retorna o resto da divisão entre dois números (%). Utilizando o mesmo exemplo anterior, quando dividimos 11 por 2, temos como resultado inteiro 5 e resto 1. Por que resto 1? Note que 11 pode ser representado como 11 = 5\*2 + 1, tendo o 1 como resto da divisão. Observe o exemplo abaixo:

In [31]:
# Atribuindo o valor 11 à variável x
x = 11

# Atribuindo o valor 2 à variável y
y = 2

# Exibindo o valor do resto da divisão de x por y
print(f"O resto da divisão de x por y é: {x%y}")

O resto da divisão de x por y é: 1


<hr>

# 3. Formatação de números em Python
Quando vamos exibir números em Python, como fizemos acima, muitas vezes queremos formatá-los de uma maneira específica. Por exemplo, podemos querer exibir um número com 2 casas decimais, 3 casas decimais ou com separador de milhares para torná-lo mais fácil de ler. Para atingir esse objetivo, podemos usar as operações de formatação de números em Python.

## Formatação de casas decimais
Para começar, vamos falar sobre a formatação de casas decimais. Observe o exemplo abaixo:

In [32]:
# Atribuindo o valor 3.14159 à variável x
x = 3.14159

# Exibindo x com apenas 2 casas decimais
print(f"O valor de x com duas casas decimais é: {x:.2f}")

O valor de x com duas casas decimais é: 3.14


Note que o número foi exibido com apenas duas casas decimais, fazendo o arredondamento matemático. Para provar, observe o outro exemplo abaixo:

In [33]:
# Atribuindo o valor 10.56 à variável 'numero'
numero = 10.56

# Exibindo numero com 1 casa decimal
print(f"O valor de 'numero' com uma casa decimal é: {numero:.1f}")

O valor de 'numero' com uma casa decimal é: 10.6


Note que ele arredondou para 10.6, visto que 0.56 é maior ou igual a 0.5.

## Formatação de separador de milhares
Agora, vamos falar sobre a formatação com separador de milhares no Python. Lembre-se de que a base da linguagem é o inglês e lá eles utilizam a vírgula (,) para separar os milhares e o ponto (.) para separar os decimais, ao contrário de nós. Observe o exemplo abaixo:

In [34]:
# Atribuindo o valor 1947 à variável num
num = 1947

# Exibindo num com separador de milhares
print(f"O valor de num separado por milhar é: {num:,}")

O valor de num separado por milhar é: 1,947


Agora, vamos fazer um exemplo separando milhares e filtrando as casas decimais.

In [35]:
# Atribuindo o valor 1875.8965 à variável num
num = 1875.8965

# Exibindo num com separador de milhares e com 2 casas decimais
print(f"O valor de 'num' duplamente formatado é: {num:,.2f}")

O valor de 'num' duplamente formatado é: 1,875.90


Essas são algumas das formatações disponíveis no Python. Tem outras para strings e datas, mas não acho necessário falar sobre elas agora. Para finalizar, vamos falar sobre ordem de precedência. 

<hr>

# 4. Ordem de precedência
Você já viu aqueles desafios que viralizam na internet? Observe o exemplo de desafio:

<img src="https://i.pinimg.com/236x/53/ae/fa/53aefa58d82fccfefeb48fa5c452bfee.jpg" class="center" width="300">

Muitas pessoas erram esse tipo de problema porque não sabem sobre o tema que vamos abordar agora. A ordem de precedência é a ordem em que os operadores são executados em uma expressão. Em Python, a ordem de precedência é a seguinte:

<ol>
    <li> Parênteses
    <li> Potência (**)
    <li> Multiplicação (*), divisão (/), divisão inteira (//) e resto da divisão (%)
    <li> Adição (+) e subtração (-)
</ol>

O exemplo acima, por exemplo, temos como resultado 10, porque realizamos primeiro as operações de multiplicação (5\*2) e (1\*0) e, somente depois, realizamos a adição dos resultados das multiplicações (10 + 0), que resulta em 10.

Agora, segue uma implementação em Python:

In [36]:
# Expressão do exemplo
expressao = 5*2 + 1*0

# Exibindo o resultado
print(f"O resultado da expressão é: {expressao}")

O resultado da expressão é: 10


In [38]:
# Atribuindo o valor 5 à variável n1
n1 = 5

# Atribuindo o valor 11 à variável n2
n2 = 11

# Calculando a média desses números
media = (n1+n2)/2

# Exibindo resultado
print(f"O resultado da média é: {media}")

O resultado da média é: 8.0


# Exercícios
Chegou a hora de colocar seus conhecimentos em prática!

## Exercício 009
Faça um programa que peça ao usuário para digitar dois números e mostre na tela o resultado da soma, subtração, multiplicação, divisão e resto da divisão desses números

## Exercício 010
Faça um programa que peça ao usuário o raio de um círculo e exiba na tela a área e o perímetro desse círculo (considere pi = 3.14). Aproxime para 2 casas decimais

## Exercício 011
Faça um programa que peça ao usuário o preço de um produto e exiba o preço com um desconto de 10%

## Exercício 012
Faça um programa que leia a temperatura em graus Celsius e exiba a temperatura em graus Fahrenheit. A fórmula para converter de Celsius para Fahrenheit é: F = (9/5)\*C + 32

## Exercício 013
Faça um programa que peça ao usuário para digitar 3 números inteiros e exiba a média aritmética desses números. Aproxime para 1 casa decimal.

## Exercício 014
Faça um programa que leia o peso e a altura de uma pessoa e exiba o índice de massa corporal (IMC) dela. A fórmula para calcular o IMC é: IMC = peso/altura², com aproximação em 3 casas decimais.

## Exercício 015
Faça um programa que leia dois números inteiros do usuário e troque seus valores, ou seja, se o primeiro número for 5 e o segundo número for 7, o programa deve fazer com que o primeiro número seja igual a 7 e o segundo número seja igual a 5.

## Exercício 016
Escreva um programa Python que leia um número inteiro e verifique se ele é par ou ímpar.

## Exercício 017
Escreva um programa Python que leia um número e retorne seu quadrado, sua raíz quadrada e sua raíz cúbica, com aproximação em 2 casas decimais.

## Exercício 018
Escreva um programa Python que leia o valor de dois catetos e retorne o valor da hipotenusa, assumindo que seja possível formar um triângulo